### DISCUSSION TITLE
Converting from Colab notebooks to CHTC

### DISCUSSION THREAD

**Discussion by agitter** [CHTC/templates-GPUs#15] [General]
This discussion will track ideas for converting Colab notebooks to be runnable in CHTC.  We'll first identify issues that need to be solved and then possible solutions.

**Comment by PatStiles** [comment]
Some Links I found regarding using google collab on a local runtime:

1.) https://research.google.com/colaboratory/local-runtimes.html
This outlines running a google collab file locally by instantiating a jupyter runtime environment on a server

2.) https://stackoverflow.com/questions/53514243/google-colab-local-runtime-use
This thread outlines the similarities/differences of using a local runtime for collab/jupyter notebooks compared to using browser based.

3.) https://dev.to/ikeyasu/how-to-use-google-colaboratory-with-local-runtime-4j1p
example of setting up a nvidia docker image for google collab use and mounting it runtime

**Comment by agitter** [comment]
We wanted to discuss ways to turn a notebook into an executable for a batch job.  One way is to use a Jupyter command in a bash script that would run the notebook directly.  My group has used that strategy for continuous integration testing on notebooks (where we want to keep the original file in the notebook format): https://github.com/gitter-lab/nn4dms/blob/ee3c9c5bf1a3094b10b2add5baec9eeaaec82fd5/.github/workflows/test.yml#L29

Let's think through the pros and cons of running the notebook directly versus exporting it as a Python script and running that in the executable.

@PatStiles that [colab-local](https://hub.docker.com/r/ikeyasu/colab-local/) Docker image is promising.  The latest version was pushed 3 years ago, so we most likely couldn't use that specific one directly.  We could modify and update it or search DockerHub for others that are actively maintained.
